server/game/cards/01_SOR/units/TheGhostSpectreHomeBase.ts

```diff
@@ -0,0 +1,29 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { Trait } from '../../../core/Constants';
+
+export default class TheGhostSpectreHomeBase extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '6931439330',
+            internalName: 'the-ghost#spectre-home-base'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Give a shield token to another Spectre unit',
+            when: {
+                onAttackDeclared: (event, context) => event.attack.attacker === context.source,
+                onCardPlayed: (event, context) => event.card === context.source
+            },
+            targetResolver: {
+                optional: true,
+                cardCondition: (card, context) => card !== context.source && card.isUnit() && card.hasSomeTrait(Trait.Spectre),
```

Comment:
Card.isUnit() technically isn't needed here since only units can host upgrades, so the `giveShield()` targeting will automatically filter them out